src/unix/os390-syscalls.c

```diff
@@ -332,3 +332,54 @@ char* mkdtemp(char* path) {
 
   return path;
 }
+
+
+ssize_t os390_readlink(const char* path, char* buf, size_t len) {
+  ssize_t rlen;
+  ssize_t vlen;
+  ssize_t plen;
+  char* delimiter;
+  char old_delim;
+  char tmpbuf[len + 1];
```

Comment:
Is there a risk of stack overflows if `len` is big?

Reply:
Yes, there is a risk. I will allocate on heap instead